container/lxd/initialisation_linux.go

```diff
@@ -79,24 +79,53 @@ func (ci *containerInitialiser) Initialise() error {
 	if err := ensureDependencies(ci.lxdSnapChannel, localSeries); err != nil {
 		return errors.Trace(err)
 	}
-	err = ci.configureLxdBridge()
-	if err != nil {
-		return errors.Trace(err)
-	}
 	proxies := proxy.DetectProxies()
 	err = ci.configureLxdProxies(proxies, ci.isRunningLocally, ci.newLocalServer)
 	if err != nil {
 		return errors.Trace(err)
 	}
 
-	output, err := ci.getExecCommand(
-		"lxd",
-		"init",
-		"--auto",
-	).CombinedOutput()
+	var output []byte
+	if ci.containerNetworkingMethod == "local" {
+		output, err = ci.getExecCommand(
+			"lxd",
+			"init",
+			"--auto",
+		).CombinedOutput()
 
-	if err == nil {
-		return nil
+		if err == nil {
+			return nil
+		}
+
+	} else {
+
+		lxdInitCfg := `config: {}
+networks: []
+storage_pools:
+- config: {}
+  description: ""
+  name: default
+  driver: dir
```

Comment:
This prevents users from priming LXD and using ZFS (or any other storage backend). We end up with errors like:

```bash
2024-09-17 13:33:04 ERROR juju.worker.dependency engine.go:695 "lxd-container-provisioner" manifold worker returned unexpected error: setting up container dependencies on host machine: running lxd init: Error: Storage pool "default" is of type "zfs" instead of "dir"
```

This means that any LXD container backup process will potentially take hours due to not being able to do CoW snapshots of the container.

Will open an issue as well, to potentially discuss a way forward.